script.js

```diff
@@ -44,4 +56,199 @@ hamburgerBtn.addEventListener('click', displayMobileMenu);
+  const actionButton = document.createElement('button');
+  actionButton.classList.add('btn-max-width', 'open-popup-btn');
+  actionButton.innerText = 'See Project';
+  workCard.appendChild(actionButton);
+
+  // show action button and change background on hover
+  workCard.addEventListener('mouseenter', () => {
+    actionButton.classList.add('show');
+    workCard.style.background = `url(${hoverImageLink})`;
+    workCard.style.backgroundRepeat = 'no-repeat';
+    workCard.style.backgroundSize = 'cover';
+  });
+  // hide action button when not hovering
+  workCard.addEventListener('mouseleave', () => {
+    actionButton.classList.remove('show');
+    workCard.style.background = `url(${imageLink})`;
+    workCard.style.backgroundRepeat = 'no-repeat';
+    workCard.style.backgroundSize = 'cover';
+  });
+});
+
+/// popup window variables
+const popupWindow = document.querySelector('#popup-window');
+const popupBackground = document.querySelector('#popup-bg');
+const closePopupBtn = document.querySelector('.close-popup-btn');
+const hiddenSection = document.querySelector('#main-section');
+// popup window elements
+const projectTitle = document.querySelector('.project-title');
+const popupTechnologiesContainer = document.querySelector('#tag-popup');
+const popupImage = document.querySelector('.project-placeholder-img.popup');
+const popupDescription = document.querySelector('.project-description');
+// creat popup window desktop version
+function openPopUpDesktop(id) {
+  // Add content to popup
+  projectTitle.innerText = projectDetail[id].name;
+  const technoPopup = Object.keys(projectDetail[id].technologies);
+  technoPopup.forEach((btn) => {
+    const popupTechnologies = document.createElement('ul');
+    popupTechnologies.classList.add('card-tags');
+    const popupTechnologiesItem = document.createElement('li');
+    const popupTechnologiesBtn = document.createElement('button');
+    popupTechnologiesBtn.classList.add('tag-popup');
+    popupTechnologiesBtn.innerText = `${projectDetail[id].technologies[btn]} `;
+    popupTechnologiesItem.appendChild(popupTechnologiesBtn);
+    popupTechnologies.appendChild(popupTechnologiesItem);
+    popupTechnologiesContainer.appendChild(popupTechnologies);
+  });
+  popupImage.src = `${projectDetail[id].featured_image} `;
+  popupDescription.innerText = projectDetail[id].description;
+  window.scrollTo(0, 0);
+  popupWindow.style.display = 'block';
+  popupBackground.classList.add('popup-bg');
+  hiddenSection.style.display = 'none';
+}
+function closePopUpDesktop() {
+  window.location.href = '#portfolio';
+  popupWindow.style.display = 'none';
+  popupBackground.classList.remove('popup-bg');
+}
+if (desktopScreen.matches) {
+  const workCard = document.querySelectorAll('.work-card');
+  workCard.forEach((card) => {
+    card.addEventListener('click', (e) => {
+      e.preventDefault();
+      const id = e.target.parentNode.getAttribute('data-id');
+      openPopUpDesktop(id);
+    });
+  });
```

Comment:
- Great Job adding the modal popup to your portfolio, but I guess there may be a bug causing all project cards to disappear when the `close` button is clicked on. 
Here is what I mean 👇🏽 

| After clicking the close button on the popup  |
| :---:   | 
|<img width="400" alt="Screenshot 2023-01-05 at 18 05 18" src="https://user-images.githubusercontent.com/99246359/210898497-ccad0053-b721-4f77-b54e-18b93e6d4167.png">  |
⚠️ Kindly note that this happens on both mobile and desktop screens. 

- Also, I think some content of the modal popup on mobile isn't being displayed. Can we try to stick to Figma as much as possible? 🤗 

This is what I observed from my end 👇🏽 

| Your mobile popup  | That of Figma |
| :---:   | :---: | 
| <img width="396" alt="Screenshot 2023-01-05 at 18 04 41" src="https://user-images.githubusercontent.com/99246359/210899058-0b4cdf80-60db-4f56-9d86-c6928f81e5d7.png"> |<img width="321" alt="Screenshot 2023-01-05 at 18 25 09" src="https://user-images.githubusercontent.com/99246359/210899201-ee3fc209-1206-4417-b9c1-ff61e940ebc6.png">  |

Reply:
Thank you for your feedback. As per the comments I have made the following changes,

-  fixed the bug that hides the work cards when the popup window gets closed
- I have styled the mobile version of the popup window